### PR TITLE
chore(crouch): update crouch and crawl

### PR DESCRIPTION
**Pull request by MadsLeander**
Add updated code from the original repository as requested by @alberttheprince 

This update does not change the functionality but should fix some small bugs and issues.

This has only been briefly tested, more testing is needed before this pr is accepted.

### CODE REVIEW

**Comment by iSentrie** [comment]
The only issue for me was registered key binds `crouch` / `crawl` which didn't allowed me to crouch, it immediately canceled crouching which made me stand back up and `crawl` didn't allowed me to stand up. I changed commands to other name and then it worked properly, I guess multiple keybinds on same button was in affect. Not sure if we can call it as script issue or controls/binds of FiveM itself. Any ideas for solution?

Other weird thing is that when prone type is changed, controls inverts, but it's understandable, might be changed or kept this way.

In conclusion everything seems okay and could be merged.

**Comment by alberttheprince** [comment]
I think the prone control changes make sense if you think about the character turning around. Maybe a camera change can be added later to indicate something like this to the player if it's confusing? 

Although make sense from the perspective of changing how you're crawling around since you're technically facing the opposite direction.

**Comment by alberttheprince** [comment]
Additionally with this issue:

>The only issue for me was registered key binds crouch / crawl which didn't allowed me to crouch, it immediately canceled crouching which made me stand back up

Is this when crouch/crawl are on the same keybind?

**Comment by iSentrie** [comment]
> Additionally with this issue:
> 
> > The only issue for me was registered key binds crouch / crawl which didn't allowed me to crouch, it immediately canceled crouching which made me stand back up
> 
> Is this when crouch/crawl are on the same keybind?

No, it's some weird FiveM bug, I think. So basically for some reason keybinds gets replicated, when command will work just fine, like /crouch will work just fine. Unbinding control and binding it again fixes this issue. Why exactly it duplicates I'm not so sure.

**Comment by iSentrie** [comment]
> I think the prone control changes make sense if you think about the character turning around. Maybe a camera change can be added later to indicate something like this to the player if it's confusing?
> 
> Although make sense from the perspective of changing how you're crawling around since you're technically facing the opposite direction.

Yup, its even more interesting this way.
